### PR TITLE
Add a nil check for s3 keys

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: src/witan/app/data.clj [hunk 1]
```diff
@@ -119,16 +119,19 @@
...
-                                 :category category
-                                 :name name
-                                 :file-name file-name
-                                 :publisher publisher
-                                 :version version
-                                 :public? public?
-                                 :s3-key s3-key} %)) '(:data_by_data_id :data_by_category :data_by_s3_key))
-    (c/exec (update-version-number-name name version))
-    (get-data-by-data-id data-id)))
+    (if s3-key
```
Comment (otfrom) [inline src/witan/app/data.clj]: perhaps use schema to make sure you have everything you want here?

**Comment by elisehuard** [comment]
if we really wanted to check integrity, we'd check for s3 existence, but this does avoid the previous issue! :shipit:

File: src/witan/app/data.clj (hunk 1 above)
Comment (elisehuard) [inline src/witan/app/data.clj]: ah, good idea

**Comment by elisehuard** [comment]
cool, this will avoid nils everywhere. if tests pass :shipit:

**Pull request merged**